File: Data Structures & Algorithms/is-subsequence/submission-1.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s:
            return True
        if not t:
            return False
        i = 0
        cur = s[i]
        for l in t:
            print(i)
            print(s[i])
            if l == cur:
                i += 1
                if i == len(s):
                    return True
                cur = s[i]
            if i == len(s):
                print("here!")
                return True
        return False

File: Data Structures & Algorithms/middle-of-the-linked-list/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int n = 0;
        ListNode* cur = head;
        while(cur){
            n++;
            cur = cur->next;
        }
        n/=2;
        cur = head;
        while(n){
            cur = cur->next;
            n--;
        }
        return cur;
    }
};